scripts/compactor_runner.py

```diff
@@ -357,22 +329,118 @@ def _complete_config(self, args):
             config.startCheckpointing = args.startCheckpointing
         return config
 
+    def _configure_server_address(self, compactor_config):
+        """
+        Configure Server hostname and port from args and config file.
+        This method invokes resolve ip method to configure from network interfaces
+        when the hostname is not present in args or config file.
+
+        :param compactor_config: values parsed from config file
+        :return:
+        """
+        # configuring hostname and port
+        pod_name = os.environ.get(POD_NAME, "default_name")
+        pod_namespace = os.environ.get(POD_NAMESPACE, "default_namespace")
+        if pod_name != "default_name" and pod_namespace != "default_namespace":
+            self._config.hostname = pod_name + ".corfu-headless." + pod_namespace + ".svc.cluster.local"
+
+        server_address = compactor_config["ServerAddress"]
+        if not self._config.hostname:
+            if self._config.network_interface and self._config.network_interface_version:
+                self._print_and_log("WARNING: _configure_server_address: Couldn't load the hostname from args."
```

Comment:
nit: Hostname not passed in args.

Reply:
Done, Thank you. Warning because ideally, the hostname should be there. Self-IP discovery is not preferred.

Reply:
I see... in that case if hostname is not passed through args, do we want to go ahead with the hostname in the config file first if that's more preferred? Just trying to see what's the order of priority since there are so many options.

Reply:
Cool. Args are preferred always. Order -
1. Args Hostname
2. Args NetworkInterface and NetworkInterfaceVersion
3. Config file Hostname
4. Config file NetworkInterface and NetworkInterfaceVersion